src/main/java/ch/njol/skript/expressions/ExprFunction.java

```diff
@@ -0,0 +1,131 @@
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.function.DynamicFunctionReference;
+import ch.njol.skript.lang.function.Functions;
+import ch.njol.skript.lang.function.Namespace;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.registrations.Feature;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.lang.script.Script;
+
+import java.io.File;
+import java.util.Objects;
+
+@Name("Function (Experimental)")
+@Description("Obtain a function by name, which can be executed.")
+@Examples({
+		"set {_function} to the function named \"myFunction\"",
+		"run {_function} with arguments 13 and true"
+})
+@Since("INSERT VERSION")
+@SuppressWarnings("rawtypes")
+public class ExprFunction extends SimpleExpression<DynamicFunctionReference> {
+
+	static {
+		Skript.registerExpression(ExprFunction.class, DynamicFunctionReference.class, ExpressionType.COMBINED,
+				"[the|a] function [named] %string% [local:(in|from) %-script%]",
+				"[the] functions [named] %strings% [local:(in|from) %-script%]",
+				"[all [[of] the]|the] functions (in|from) %script%"
+		);
+	}
+
+	private Expression<String> name;
+	private Expression<Script> script;
+	private int mode;
+	private boolean local;
+	private Script here;
+
+	@Override
+	@SuppressWarnings("null")
+	public boolean init(Expression<?>[] expressions, int matchedPattern, Kleenean isDelayed,
+                        ParseResult result) {
+		if (!this.getParser().hasExperiment(Feature.SCRIPT_REFLECTION))
+			return false;
+		this.mode = matchedPattern;
+		this.local = result.hasTag("local") || mode == 2;
+		switch (mode) {
+			case 0, 1 -> {
+				//noinspection unchecked
+				this.name = (Expression<String>) expressions[0];
+				if (local)
+					//noinspection unchecked
+					this.script = (Expression<Script>) expressions[1];
+			}
+			case 2 ->
+				//noinspection unchecked
+				this.script = (Expression<Script>) expressions[0];
+		}
+		this.here = this.getParser().getCurrentScript();
+		return true;
+	}
+
+	@Override
+	protected DynamicFunctionReference<?>[] get(Event event) {
+		@Nullable Script script;
+		if (local) {
+			script = this.script.getSingle(event);
+		} else {
+			script = here;
+		}
+		return switch (mode) {
+			case 0:
+				@Nullable String name = this.name.getSingle(event);
+				if (name == null)
+					yield CollectionUtils.array();
+				@Nullable DynamicFunctionReference reference = DynamicFunctionReference.resolveFunction(name, script);
+				if (reference == null)
+					yield CollectionUtils.array();
+				yield CollectionUtils.array(reference);
+			case 1:
+				yield this.name.stream(event).map(string -> DynamicFunctionReference.resolveFunction(string, script))
+						.filter(Objects::nonNull)
+						.toArray(DynamicFunctionReference[]::new);
```

Comment:
if you used the arrow label, you could avoid the yield words (and just surround the other cases in brackets)